addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/CORSActiveScanRule.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2020 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+
+/**
+ * The CORS active scan rule identifies Cross-Origin Resource Sharing (CORS) support and overly
+ * lenient or buggy implementations
+ *
+ * @author CravateRouge
+ */
+public class CORSActiveScanRule extends AbstractAppPlugin {
+    private static Logger LOG = Logger.getLogger(CORSActiveScanRule.class);
+    private static final String RANDOM_NAME = RandomStringUtils.random(8, true, true);
+
+    @Override
+    public void scan() {
+        HttpMessage baseMsg = getBaseMsg();
+        URI uri = baseMsg.getRequestHeader().getURI();

```

Comment:
```suggestion
        URI uri = getBaseMsg().getRequestHeader().getURI();
```

No point doing an intermediate store for a single operation.